src/scheduler/limits.c

```diff
@@ -2913,6 +2913,18 @@ clear_limres(void)
 	free_resource_list(limres);
 	limres = NULL;
 }
+/**
+ *      @brief returns a linked list of resources being limited.
+ *
+ *      @par This is to be treated as read-only.  Modifying this will adversely
+ *           affect the limits code
+ *
+ *      @return schd_resource *
+ */
+schd_resource *query_limres()
```

Comment:
I'm not sure why this function is needed because it just returns a global pointer, but if we do, then should we use it in limits.c as well when we iterate through all the limit resources.

Reply:
This is a weirdity that I tried to cover in my PR description.  The limits code is treated more as a "library" to the scheduler even though it's in the scheduler directory.  It's header file lives in include instead of scheduler.  The rest of the scheduler only calls functions which are provided by the limits interface.  This is why I chose to go this way instead of just removing the static from limres.  The limits code itself is inside the "library" so it can directly use a private member.  I'd really rather not have to query this member at all.  It's needed by the limits code and if it is changed, the limits code will not function properly.

Reply:
Ok I get it now. Thanks!